### DISCUSSION TITLE
Defining a Robin boundary condition in a 1D problem

### DISCUSSION THREAD

**Discussion by amnchacon** [firedrakeproject/firedrake#3379] [Firedrake support]
Hello everyone. I'm trying to solve a 1D heat conduction problem considering a Robin boundary condition (RBC). The boundary value problem is:

$-k(u_x)_x + \sigma u = f$

$x \in (a,b)$

with $u(a) = u_a$

and $-ku_x(b) = \alpha[u(b) - u_\infty]$ --> RBC

The variational formulation after integrate by parts is:

$\int_a^b (ku_xv_x + \sigma uv) dx + \alpha u(b)v(b) = \int_a^b fv dx + \alpha u_\infty v(b)$, $\forall v \in \mathcal{V}$

The parameters $k$, $\sigma$, $\alpha$, $f$ and $u_\infty$ are constants. 

1. How can I write correctly this variational formulation in Firedrake?

a) My problem is not about how to write the integral terms, but the terms $\alpha u(b)v(b)$ (at the left hand side) and $\alpha u_\infty v(b)$ (at the right hand side). Notice that those terms are not integrals, $u(b)$ is a variable at the boundary ($x=b$), and $v(b)$ is related to $v \in \mathcal{V}$. How write those terms in the code?

b) The writing of those not integral terms would change depending on the type of mesh, Unit Interval Mesh, Square Mesh or Annulus Mesh? 

My gratitude...

**Comment by ksagiyam** [comment]
Those are terms coming from the application of divergence theorem, and those are indeed surface integrals, and you could still use `ds` integrals to represent them as suggested in https://github.com/firedrakeproject/firedrake/discussions/3369.

**Reply by amnchacon** [reply]
Hello @ksagiyam. Thanks. But this is a 1D case. In 2-3D cases those integrals can work well. In fact, I built a 2D case thinking in the same 1D case and it worked. It seems (I imagine) that Firedrake has not implemeted these type of cases (when you're working with a Robin boundary condition in a 1D case). My point is that if you have a Unit interval mesh, and after integrate by parts you have $[vu]_0^1$, this can't be tretated as a surface integral like v * u * ds, because is not an integral; it's a term emerged after integrate. In 2D or 3D cases is diferent, because those surface integrals, like:

$\int_{d\Omega_N} qv d\Omega_N$

Emerges naturally after integrate. It was what I treated to say to @colinjcotter in that discussion. One thing is when you (in a 2D or 3D case) write in Firedrake the integral above: q * v * ds (or ds_t, ds_b, dS... etc, depending on the geometry mesh you're working), but another thing is when in a 1D case, you have individual terms like $\alpha u(b)v(b)$, $\alpha u_\infty v(b)$....

**Reply by dham** [reply]
This is not correct. The 1D case is the same as the 2D or 3D cases. In 1D the boundary facets are points and the boundary integrals that result from integrating by parts are integrals against a point measure. This is equivalent to evaluating the integrand at that point and therefore makes the boundary integral equivalent to the point evaluation that is usually written when integrating by parts on a 1D domain.

**Reply by amnchacon** [reply]
Oh, I see. I thought the software internally integrate for a second time those terms that in 1D case emerged after integrate. But, I have a case is not working. Well, I'll review carefully. Thanks for your replying.

**Comment by colinjcotter** [comment]
No, there is no magic. It just does the integrals that you ask for.

From: Andrés Mauricio Nieves Chacón ***@***.***>
Date: Friday, 2 February 2024 at 17:12
To: firedrakeproject/firedrake ***@***.***>
Cc: Cotter, Colin J ***@***.***>, Mention ***@***.***>
Subject: Re: [firedrakeproject/firedrake] Defining a Robin boundary condition in a 1D problem (Discussion #3379)

Oh, I see. I thought the software internally integrate for a second time those terms that in 1D case emerged after integrate. But, I have a case is not working. Well, I'll review carefully. Thanks for your replying.
